.github/workflows/main.yaml

```diff
@@ -0,0 +1,59 @@
+name: Build and Test
+
+# This workflow will run on main branch and on any pull requests targeting main
+on:
+  push:
+    branches:
+      - main
+    tags: []
+  pull_request:
+  
+jobs:
+  lint:
+    name: Lint
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set up Go
+        uses: actions/setup-go@v1
+        with:
+          go-version: 1.17
+
+      - name: Check out code
+        uses: actions/checkout@v1
+
+      - name: Lint Go Code
+        run: |
+          export PATH=$PATH:$(go env GOPATH)/bin # temporary fix. See https://github.com/actions/setup-go/issues/14
+          go get -u golang.org/x/lint/golint 
```

Comment:
This should use `golangci-lint`.

https://github.com/golangci/golangci-lint-action